pkg/views/server/configure.go

```diff
@@ -113,23 +134,75 @@ func ConfigurationForm(config *apiclient.ServerConfig, containerRegistries []api
 		huh.NewGroup(
 			huh.NewInput().
 				Title("Frps Domain").
-				Value(config.Frps.Domain),
+				Value(m.config.Frps.Domain),
 			huh.NewInput().
 				Title("Frps Port").
 				Value(&frpsPortView).
-				Validate(createPortValidator(config, &frpsPortView, config.Frps.Port)),
+				Validate(createPortValidator(m.config, &frpsPortView, m.config.Frps.Port)),
 			huh.NewInput().
 				Title("Frps Protocol").
-				Value(config.Frps.Protocol),
+				Value(m.config.Frps.Protocol),
 		),
-	).WithTheme(views.GetCustomTheme())
+	)
+}
+
+func (m Model) Init() tea.Cmd {
+	return m.form.Init()
+}
+
+func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+s":
+			m.quiting = true
+			return m, tea.Quit
+		case "ctrl+c":
+			m.config = nil
+			return m, tea.Quit
+		}
+	}
+
+	var cmds []tea.Cmd
+	form, cmd := m.form.Update(msg)
+	if f, ok := form.(*huh.Form); ok {
+		m.form = f
+		cmds = append(cmds, cmd)
+	}
+
+	if m.form.State == huh.StateCompleted {
+		m.quiting = true
+		return m, tea.Quit
+	}
+
+	return m, tea.Batch(cmds...)
+}
+
+func (m Model) View() string {
+	if m.quiting {
+		return ""
+	}
+
+	helpView := m.help.ShortHelpView([]key.Binding{
+		m.keymap.submit,
+	})
+	return m.form.View() + "  " + helpView
+}
+
+func ConfigurationForm(config *apiclient.ServerConfig, containerRegistries []apiclient.ContainerRegistry) (*apiclient.ServerConfig, error) {
+	m := NewModel(config, containerRegistries)
+
+	p, err := tea.NewProgram(m).Run()
 
-	err := form.Run()
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
+	}
+
+	if m, ok := p.(Model); ok && m.config != nil {
+		return m.config, nil
 	}
 
-	return config
+	return nil, errors.New("failed to get server config")
```

Comment:
Can we just return nil instead of a fatal error in case the user "ctrl-c"-s?
Also the message in case something really went wrong should mention the configuration form instead of "getting the server config"

Reply:
errors.New("No changes were made.")

We could use this error message instead, as we need it to ensure the user has aborted the form.

Reply:
Okay do it, #800 takes care of ctrl-c-ing either way so I will let the other contributor to take a look at that

Reply:
ok